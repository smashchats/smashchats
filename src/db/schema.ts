import { relations, sql, eq, desc } from "drizzle-orm";
import {
    integer,
    sqliteTable,
    sqliteView,
    text,
} from "drizzle-orm/sqlite-core";
import { Contact } from "@/src/models/Contacts";
import { Message } from "@/src/models/Messages";
import { TrustRelation } from "@/src/models/TrustRelation";

const timestamp = (fieldName: string) =>
    integer(fieldName, { mode: "timestamp" });

const defaultTimestamp = (fieldName: string) =>
    timestamp(fieldName)
        .default(sql`(unixepoch())`)
        .notNull();

const boolean = (fieldName: string, defaultValue: boolean = false) =>
    integer(fieldName, { mode: "boolean" }).default(defaultValue);

export const contacts = sqliteTable("contacts", {
    did_id: text("did_id", { mode: "text" }).primaryKey(),
    did_ik: text("did_ik", { mode: "text" }).notNull(),
    did_ek: text("did_ek", { mode: "text" }).notNull(),
    did_signature: text("did_signature", { mode: "text" }).notNull(),
    did_endpoints: text("did_endpoints", { mode: "json" }).notNull(),
    is_self: boolean("is_self"),
    notes: text(),
    meta_title: text(),
    meta_description: text(),
    meta_picture: text(),
    scores: text("scores", { mode: "json" }),
    smashed: boolean("smashed"),
    blocked: boolean("blocked"),
    created_at: defaultTimestamp("created_at"),
    updated_at: defaultTimestamp("updated_at"),
});

export const trustRelations = sqliteTable("trust_relations", {
    did_id: text("did_id", { mode: "text" }).primaryKey(),
    created_at: defaultTimestamp("created_at"),
    name: text("name", { mode: "text" }).notNull(),
});

export const messages = sqliteTable("messages", {
    sha256: text().primaryKey(),
    timestamp: defaultTimestamp("timestamp"),
    type: text("type", { mode: "text" }).notNull(),
    data: text("data", { mode: "text" }).notNull(),
    after_sha256: text(),
    reply_to_sha256: text(),
    from_did_id: text()
        .notNull()
        .references(() => contacts.did_id),
    discussion_id: text()
        .notNull()
        .references(() => contacts.did_id),
    created_at: defaultTimestamp("created_at"),
    date_delivered: timestamp("date_delivered"),
    date_read: timestamp("date_read"),
});

export const contactsRelations = relations(contacts, ({ many }) => ({
    messages: many(messages),
}));

export const messagesRelations = relations(messages, ({ one }) => ({
    from: one(contacts, {
        fields: [messages.from_did_id],
        references: [contacts.did_id],
    }),
    to: one(contacts, {
        fields: [messages.discussion_id],
        references: [contacts.did_id],
    }),
}));

export const trustRelationsRelations = relations(trustRelations, ({ one }) => ({
    contact: one(contacts, {
        fields: [trustRelations.did_id],
        references: [contacts.did_id],
    }),
}));

export const chatListView = sqliteView("chat_list_view").as((qb) =>
    qb
        .select({
            did_id: contacts.did_id,
            meta_title: contacts.meta_title,
            meta_picture: contacts.meta_picture,
            smashed: contacts.smashed,
            most_recent_message: messages.data,
            most_recent_message_type: messages.type,
            trusted_name: trustRelations.name,
            created_at: contacts.created_at,
            most_recent_message_date:
                sql<number>`MAX(${messages.created_at})`.as(
                    "most_recent_message_date"
                ),
            unread_count:
                sql<number>`COUNT(${messages.sha256}) - COUNT(${messages.date_read})`.as(
                    "unread_count"
                ),
        })
        .from(contacts)
        .leftJoin(messages, eq(contacts.did_id, messages.discussion_id))
        .leftJoin(trustRelations, eq(contacts.did_id, trustRelations.did_id))
        .groupBy(contacts.did_id)
        .orderBy(desc(messages.created_at))
);

export interface ChatListView {
    did_id: Contact["did_id"];
    meta_title: Contact["meta_title"];
    meta_picture: Contact["meta_picture"];
    smashed: Contact["smashed"];
    created_at: Contact["created_at"];
    most_recent_message: Message["data"];
    most_recent_message_type: Message["type"];
    trusted_name?: TrustRelation["name"];
    most_recent_message_date: number;
    unread_count: number;
}
