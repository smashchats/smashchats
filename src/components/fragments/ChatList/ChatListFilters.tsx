import React from "react";
import { Pressable } from "react-native";
import { Box } from "@/src/components/design-system/Box.jsx";
import { Badge } from "@/src/components/design-system/Badge.jsx";
import { BadgeText } from "@/src/components/design-system/BadgeText.jsx";
import { MaterialCommunityIcons } from "@expo/vector-icons";
import {
    useGlobalDispatch,
    useGlobalState,
} from "@/src/context/GlobalContext.jsx";
import { NEIGHBOURHOOD_FILTERS } from "@/data/neighbourhood.js";

export const COMMON_FILTERS = ["unread", "smashed", "trusted"];

export function ChatListFilters() {
    const { chatList } = useGlobalState();
    const dispatch = useGlobalDispatch();

    const filters = [...COMMON_FILTERS, ...NEIGHBOURHOOD_FILTERS];

    return (
        <Box flexDirection="row" marginBottom={10} paddingHorizontal={10}>
            {[
                ...chatList.selectedFilters,
                ...filters.filter((f) => !chatList.selectedFilters.includes(f)),
            ].map((filter) => {
                const selected = chatList.selectedFilters.includes(filter);
                return (
                    <Pressable
                        key={filter}
                        onPress={() =>
                            dispatch({
                                type: "CHAT_LIST_TOGGLE_FILTER_ACTION",
                                filter,
                            })
                        }
                    >
                        <Badge
                            type={selected ? "smashed" : "unselected"}
                            size="lg"
                            marginRight={6}
                        >
                            {filter !== "trusted" && (
                                <BadgeText color="white">{filter}</BadgeText>
                            )}
                            {filter === "trusted" && (
                                <MaterialCommunityIcons
                                    name="check-circle"
                                    size={14}
                                    color="white"
                                />
                            )}
                        </Badge>
                    </Pressable>
                );
            })}
        </Box>
    );
}
