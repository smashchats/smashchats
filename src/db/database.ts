import { drizzle } from "drizzle-orm/expo-sqlite";
import { openDatabaseSync } from "expo-sqlite/next";
import * as schema from "@/src/db/schema.js";

export const DATABASE_NAME = __DEV__
    ? "dev-smashchats-2024-12-20-15h-40"
    : "prod-smashchats-alpha-2024-12-17";

export const expo_db = openDatabaseSync(DATABASE_NAME, {
    enableChangeListener: true,
    useNewConnection: true,
});
export const drizzle_db = drizzle(expo_db, { schema });
