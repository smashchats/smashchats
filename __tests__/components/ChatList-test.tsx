import { ChatListFilters } from "@/src/components/fragments/ChatList/ChatListFilters.jsx";
import {
    filterChatsBasedOnFilters,
    INITIAL_CHAT_LIST_STATE,
} from "@/src/context/ChatListContext.js";
import {
    GlobalParams,
    GlobalProvider,
    INITIAL_GLOBAL_STATE,
} from "@/src/context/GlobalContext.jsx";
import { ChatListView } from "@/src/db/schema";
import { fireEvent, render } from "@testing-library/react-native";

const DEFAULT_VALUES: Partial<ChatListView> = {
    most_recent_message: "string",
    most_recent_message_type: "text",
    most_recent_message_date: new Date().getTime(),
    did_id: "string",
    smashed: false,
    unread_count: 0,
};

describe("filter types", () => {
    test("unread", () => {
        const chat: ChatListView = {
            ...DEFAULT_VALUES,
            unread_count: 1,
        } as ChatListView;

        expect(filterChatsBasedOnFilters(chat, ["unread"])).toBeTruthy();
    });
    test("smashed", () => {
        const chat: ChatListView = {
            ...DEFAULT_VALUES,
            smashed: true,
        } as ChatListView;

        expect(filterChatsBasedOnFilters(chat, ["smashed"])).toBeTruthy();
        expect(filterChatsBasedOnFilters(chat, ["trusted"])).toBeFalsy();
    });
    test("unread", () => {
        const chat: ChatListView = {
            ...DEFAULT_VALUES,
            trusted_name: "trusted",
        } as ChatListView;

        expect(filterChatsBasedOnFilters(chat, ["trusted"])).toBeTruthy();
        expect(filterChatsBasedOnFilters(chat, ["smashed"])).toBeFalsy();
    });
    test("content in name", () => {
        const chat: ChatListView = {
            ...DEFAULT_VALUES,
            meta_title: "my fav cat",
        } as ChatListView;

        expect(filterChatsBasedOnFilters(chat, ["cat"])).toBeTruthy();
    });
});

describe("filter amounts", () => {
    test("one", () => {
        const chat: ChatListView = {
            ...DEFAULT_VALUES,
            unread_count: 1,
        } as ChatListView;

        expect(filterChatsBasedOnFilters(chat, ["unread"])).toBeTruthy();
    });

    test("two", () => {
        const chat: ChatListView = {
            ...DEFAULT_VALUES,
            unread_count: 2,
        } as ChatListView;

        expect(
            filterChatsBasedOnFilters(chat, ["unread", "smashed"])
        ).toBeFalsy();
        expect(
            filterChatsBasedOnFilters({ ...chat, smashed: true }, [
                "unread",
                "smashed",
            ])
        ).toBeTruthy();
    });
});

describe("empty name", () => {
    const chat: ChatListView = {
        ...DEFAULT_VALUES,
        meta_title: null,
    } as ChatListView;

    expect(filterChatsBasedOnFilters(chat, ["cat"])).toBeFalsy();
});

describe("ui", () => {
    const INITIAL_STATE = {
        ...INITIAL_GLOBAL_STATE,
        chatList: {
            ...INITIAL_CHAT_LIST_STATE,
        },
    };
    let initialState: GlobalParams;

    beforeEach(() => {
        initialState = { ...INITIAL_STATE };
    });

    it("renders correctly", () => {
        initialState.chatList.selectedFilters = ["unread", "trusted"];

        const tree = render(
            <GlobalProvider initialState={initialState}>
                <ChatListFilters />
            </GlobalProvider>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it("selects a filter when you click on it", () => {
        const { toJSON, getByText } = render(
            <GlobalProvider initialState={initialState}>
                <ChatListFilters />
            </GlobalProvider>
        );

        const filter = getByText("unread");

        expect(toJSON()).toMatchSnapshot();

        fireEvent.press(filter);

        expect(toJSON()).toMatchSnapshot();
    });
});
