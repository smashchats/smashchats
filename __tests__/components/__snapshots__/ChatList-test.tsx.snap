// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": undefined,
      "flexDirection": "row",
      "height": undefined,
      "marginBottom": 10,
      "paddingHorizontal": 10,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#64189b",
          "borderColor": "#64189b",
          "borderRadius": 16,
          "borderWidth": 2,
          "flex": undefined,
          "flexDirection": "row",
          "height": undefined,
          "justifyContent": "center",
          "marginHorizontal": undefined,
          "marginRight": 6,
          "paddingHorizontal": 8,
          "paddingVertical": 1,
          "width": undefined,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": undefined,
            "flex": undefined,
            "height": undefined,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontFamily": "Cairo, apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif",
              "fontSize": 12,
              "textTransform": "uppercase",
            }
          }
        >
          unread
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#64189b",
          "borderColor": "#64189b",
          "borderRadius": 16,
          "borderWidth": 2,
          "flex": undefined,
          "flexDirection": "row",
          "height": undefined,
          "justifyContent": "center",
          "marginHorizontal": undefined,
          "marginRight": 6,
          "paddingHorizontal": 8,
          "paddingVertical": 1,
          "width": undefined,
        }
      }
    >
      <Text>
        check-circle
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#222222",
          "borderColor": "#64189b",
          "borderRadius": 16,
          "borderWidth": 2,
          "flex": undefined,
          "flexDirection": "row",
          "height": undefined,
          "justifyContent": "center",
          "marginHorizontal": undefined,
          "marginRight": 6,
          "paddingHorizontal": 8,
          "paddingVertical": 1,
          "width": undefined,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": undefined,
            "flex": undefined,
            "height": undefined,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontFamily": "Cairo, apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif",
              "fontSize": 12,
              "textTransform": "uppercase",
            }
          }
        >
          smashed
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#222222",
          "borderColor": "#64189b",
          "borderRadius": 16,
          "borderWidth": 2,
          "flex": undefined,
          "flexDirection": "row",
          "height": undefined,
          "justifyContent": "center",
          "marginHorizontal": undefined,
          "marginRight": 6,
          "paddingHorizontal": 8,
          "paddingVertical": 1,
          "width": undefined,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": undefined,
            "flex": undefined,
            "height": undefined,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontFamily": "Cairo, apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif",
              "fontSize": 12,
              "textTransform": "uppercase",
            }
          }
        >
          👟
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#222222",
          "borderColor": "#64189b",
          "borderRadius": 16,
          "borderWidth": 2,
          "flex": undefined,
          "flexDirection": "row",
          "height": undefined,
          "justifyContent": "center",
          "marginHorizontal": undefined,
          "marginRight": 6,
          "paddingHorizontal": 8,
          "paddingVertical": 1,
          "width": undefined,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": undefined,
            "flex": undefined,
            "height": undefined,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontFamily": "Cairo, apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif",
              "fontSize": 12,
              "textTransform": "uppercase",
            }
          }
        >
          🍆
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`ui selects a filter when you click on it 1`] = `
<View
  style={
    {
      "backgroundColor": undefined,
      "flexDirection": "row",
      "height": undefined,
      "marginBottom": 10,
      "paddingHorizontal": 10,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#64189b",
          "borderColor": "#64189b",
          "borderRadius": 16,
          "borderWidth": 2,
          "flex": undefined,
          "flexDirection": "row",
          "height": undefined,
          "justifyContent": "center",
          "marginHorizontal": undefined,
          "marginRight": 6,
          "paddingHorizontal": 8,
          "paddingVertical": 1,
          "width": undefined,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": undefined,
            "flex": undefined,
            "height": undefined,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontFamily": "Cairo, apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif",
              "fontSize": 12,
              "textTransform": "uppercase",
            }
          }
        >
          unread
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#64189b",
          "borderColor": "#64189b",
          "borderRadius": 16,
          "borderWidth": 2,
          "flex": undefined,
          "flexDirection": "row",
          "height": undefined,
          "justifyContent": "center",
          "marginHorizontal": undefined,
          "marginRight": 6,
          "paddingHorizontal": 8,
          "paddingVertical": 1,
          "width": undefined,
        }
      }
    >
      <Text>
        check-circle
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#222222",
          "borderColor": "#64189b",
          "borderRadius": 16,
          "borderWidth": 2,
          "flex": undefined,
          "flexDirection": "row",
          "height": undefined,
          "justifyContent": "center",
          "marginHorizontal": undefined,
          "marginRight": 6,
          "paddingHorizontal": 8,
          "paddingVertical": 1,
          "width": undefined,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": undefined,
            "flex": undefined,
            "height": undefined,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontFamily": "Cairo, apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif",
              "fontSize": 12,
              "textTransform": "uppercase",
            }
          }
        >
          smashed
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#222222",
          "borderColor": "#64189b",
          "borderRadius": 16,
          "borderWidth": 2,
          "flex": undefined,
          "flexDirection": "row",
          "height": undefined,
          "justifyContent": "center",
          "marginHorizontal": undefined,
          "marginRight": 6,
          "paddingHorizontal": 8,
          "paddingVertical": 1,
          "width": undefined,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": undefined,
            "flex": undefined,
            "height": undefined,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontFamily": "Cairo, apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif",
              "fontSize": 12,
              "textTransform": "uppercase",
            }
          }
        >
          👟
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#222222",
          "borderColor": "#64189b",
          "borderRadius": 16,
          "borderWidth": 2,
          "flex": undefined,
          "flexDirection": "row",
          "height": undefined,
          "justifyContent": "center",
          "marginHorizontal": undefined,
          "marginRight": 6,
          "paddingHorizontal": 8,
          "paddingVertical": 1,
          "width": undefined,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": undefined,
            "flex": undefined,
            "height": undefined,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontFamily": "Cairo, apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif",
              "fontSize": 12,
              "textTransform": "uppercase",
            }
          }
        >
          🍆
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`ui selects a filter when you click on it 2`] = `
<View
  style={
    {
      "backgroundColor": undefined,
      "flexDirection": "row",
      "height": undefined,
      "marginBottom": 10,
      "paddingHorizontal": 10,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#64189b",
          "borderColor": "#64189b",
          "borderRadius": 16,
          "borderWidth": 2,
          "flex": undefined,
          "flexDirection": "row",
          "height": undefined,
          "justifyContent": "center",
          "marginHorizontal": undefined,
          "marginRight": 6,
          "paddingHorizontal": 8,
          "paddingVertical": 1,
          "width": undefined,
        }
      }
    >
      <Text>
        check-circle
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#222222",
          "borderColor": "#64189b",
          "borderRadius": 16,
          "borderWidth": 2,
          "flex": undefined,
          "flexDirection": "row",
          "height": undefined,
          "justifyContent": "center",
          "marginHorizontal": undefined,
          "marginRight": 6,
          "paddingHorizontal": 8,
          "paddingVertical": 1,
          "width": undefined,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": undefined,
            "flex": undefined,
            "height": undefined,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontFamily": "Cairo, apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif",
              "fontSize": 12,
              "textTransform": "uppercase",
            }
          }
        >
          unread
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#222222",
          "borderColor": "#64189b",
          "borderRadius": 16,
          "borderWidth": 2,
          "flex": undefined,
          "flexDirection": "row",
          "height": undefined,
          "justifyContent": "center",
          "marginHorizontal": undefined,
          "marginRight": 6,
          "paddingHorizontal": 8,
          "paddingVertical": 1,
          "width": undefined,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": undefined,
            "flex": undefined,
            "height": undefined,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontFamily": "Cairo, apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif",
              "fontSize": 12,
              "textTransform": "uppercase",
            }
          }
        >
          smashed
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#222222",
          "borderColor": "#64189b",
          "borderRadius": 16,
          "borderWidth": 2,
          "flex": undefined,
          "flexDirection": "row",
          "height": undefined,
          "justifyContent": "center",
          "marginHorizontal": undefined,
          "marginRight": 6,
          "paddingHorizontal": 8,
          "paddingVertical": 1,
          "width": undefined,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": undefined,
            "flex": undefined,
            "height": undefined,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontFamily": "Cairo, apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif",
              "fontSize": 12,
              "textTransform": "uppercase",
            }
          }
        >
          👟
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#222222",
          "borderColor": "#64189b",
          "borderRadius": 16,
          "borderWidth": 2,
          "flex": undefined,
          "flexDirection": "row",
          "height": undefined,
          "justifyContent": "center",
          "marginHorizontal": undefined,
          "marginRight": 6,
          "paddingHorizontal": 8,
          "paddingVertical": 1,
          "width": undefined,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": undefined,
            "flex": undefined,
            "height": undefined,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontFamily": "Cairo, apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif",
              "fontSize": 12,
              "textTransform": "uppercase",
            }
          }
        >
          🍆
        </Text>
      </View>
    </View>
  </View>
</View>
`;
